kevm-pyk/src/kevm_pyk/dist.py

```diff
@@ -0,0 +1,340 @@
+from __future__ import annotations
+
+import logging
+import os
+import shutil
+import time
+from argparse import ArgumentParser
+from concurrent.futures import ThreadPoolExecutor
+from contextlib import contextmanager
+from distutils.dir_util import copy_tree
+from enum import Enum
+from pathlib import Path
+from subprocess import CalledProcessError
+from tempfile import TemporaryDirectory
+from typing import TYPE_CHECKING, Literal
+
+from pyk.cli.args import KCLIArgs
+from pyk.cli.utils import loglevel
+from pyk.kbuild.utils import sync_files
+from pyk.utils import hash_str, run_process
+from xdg_base_dirs import xdg_cache_home
+
+from . import config
+from .kompile import KompileTarget, kevm_kompile
+
+if TYPE_CHECKING:
+    from argparse import Namespace
+    from collections.abc import Iterator, Mapping
+    from concurrent.futures import Future
+    from typing import Any, Final
+
+
+_LOGGER: Final = logging.getLogger(__name__)
+_LOG_FORMAT: Final = '%(levelname)s %(asctime)s %(name)s - %(message)s'
+
+
+def _dist_dir() -> Path:
+    dist_dir_env = os.getenv('KEVM_DIST_DIR')
+    if dist_dir_env:
+        return Path(dist_dir_env).resolve()
+
+    digest = hash_str({'module-dir': config.MODULE_DIR})[:7]
+    return xdg_cache_home() / f'evm-semantics-{digest}'
+
+
+DIST_DIR: Final = _dist_dir()
```

Comment:
Is this just a shim for Nix install, so it can set `KEVM_DIST_DIR`? Maybe it would be good to have a comment here about it. At first I thought we were requiring to set `KEVM_DIST_DIR` every time we called this script, but it looks like it's just for Nix.

Reply:
Done.